lib/dgram.js

```diff
@@ -272,6 +267,41 @@ Socket.prototype.send = function(buffer,
   if (offset + length > buffer.length)
     throw new RangeError('Offset + length beyond buffer length');
 
+  return buffer.slice(offset, offset + length);
+}
+
+Socket.prototype.send = function(buffer,
+                                 offset,
+                                 length,
+                                 port,
+                                 address,
+                                 callback) {
+  var self = this;
+
+  if (arguments.length === 5 || arguments.length === 6) {
+    buffer = sliceBuffer(buffer, offset, length);
```

Comment:
If you're gonna send an array of buffers, I have no idea what the point of offset and length should be. It seems a rather useless feature (hence my initial proposal for a method called `sendv` instead). I honestly have no opinion as to what the semantics here should be, but if you can find any that make any sense at all, go for it. I don't think unix's sendmsg function actually takes these arguments when sending msg_iov.